static_frame/test/unit/test_frame.py

```diff
@@ -9666,11 +9665,34 @@ def test_frame_set_index_hierarchy_f(self) -> None:
                 drop=True,
                 reorder_for_hierarchy=True,
                 )
-
         self.assertEqual(fh.to_pairs(0),
                 ((2, (((1, 'a'), 10), ((1, 'c'), 30), ((1, 'b'), 20), ((2, 'a'), 40), ((2, 'c'), 60), ((2, 'b'), 50))),)
                 )
 
+    def test_frame_set_index_hierarchy_g(self) -> None:
+        f1 = ff.parse('s(3,4)|v(dtD, dtY, str, str)')
+
+        f2 = f1.set_index_hierarchy((0, 1),
+                index_constructors=IndexAutoConstructorFactory,
+                drop=True,
+                )
+        dt64 = np.datetime64
+        self.assertEqual(f2.to_pairs(),
+                ((2, (((dt64('2210-12-26'), dt64('164167')), 'ztsv'), ((dt64('2224-04-06'), dt64('43127')), 'zUvW'), ((dt64('2202-08-20'), dt64('7699')), 'zkuW'))), (3, (((dt64('2210-12-26'), dt64('164167')), 'z2Oo'), ((dt64('2224-04-06'), dt64('43127')), 'z5l6'), ((dt64('2202-08-20'), dt64('7699')), 'zCE3')))))
+        self.assertEqual(f2.index.name, (0, 1))
+
+    def test_frame_set_index_hierarchy_h(self) -> None:
+        IACF = IndexAutoConstructorFactory
+        f1 = ff.parse('s(3,4)|v(dtD, dtY, str, str)')
+        f2 = f1.set_index_hierarchy((0, 1),
+                index_constructors=(IACF('a'), IACF('b')),
```

Comment:
The names don't populate as expected for some cases.

```python
10:58:23  In [18] f1.set_index_hierarchy((0, 1), index_constructors=(sf.IndexAutoConstructorFactory("a"), sf.IndexAutoConstructorFactory("b"))).index.name
10:59:48 Out [18] ('a', 'b')

10:59:48  In [19] f1.set_index_hierarchy((0, 1), index_constructors=(sf.IndexAutoConstructorFactory, sf.IndexAutoConstructorFactory("b"))).index.name
10:59:54 Out [19] (0, 1)

10:59:54  In [20] f1.set_index_hierarchy((0, 1), index_constructors=(sf.IndexAutoConstructorFactory("a"), sf.IndexAutoConstructorFactory)).index.name
11:00:02 Out [20] (0, 1)

```

Reply:
That is a good catch!